services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java

```diff
@@ -365,14 +367,22 @@ public News unScheduleNews(News news, org.exoplatform.services.security.Identity
    * {@inheritDoc}
    */
   public News postNews(News news, String poster) throws Exception {
-    postNewsActivity(news);
-    updateNews(news, poster, null, news.isPublished());
-    sendNotification(poster, news, NotificationConstants.NOTIFICATION_CONTEXT.POST_NEWS);
-    if (news.isPublished()) {
-      publishNews(news, poster);
-    }
-    NewsUtils.broadcastEvent(NewsUtils.POST_NEWS_ARTICLE, news.getId(), news);//Gamification
-    NewsUtils.broadcastEvent(NewsUtils.POST_NEWS, news.getAuthor(), news);//Analytics
+    News existNews;
+    try {
+      existNews = newsStorage.getNewsById(news.getId(), false);
+    } catch (ItemNotFoundException e) {
+      existNews = newsStorage.createNews(news);
+    }
```

Comment:
this method is dedicated only to post a news when it is already created. you should fix the issue on the create method cc @rdenarie

Reply:
@azayati the problem is the post and not in the create method, this fix should be okay

Reply:
create method calls post method, and here you call create method from post method. It is a cyclic call which is not permitted. You can check the existence of the news from create method before calling the post method

Reply:
> create method calls post method, and here you call create method from post method. It is a cyclic call which is not permitted. You can check the existence of the news from create method before calling the post method

i'm calling the create method in the storage to recreate the news node 
https://github.com/exoplatform/news/blob/develop/services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java#L107

Reply:
@azayati Okay i understood your point

Reply:
As I understand, we should rework the pr.
Once done, we should retest it. So please add your modification on feature/maitenance, so that QA will be able to retest

Reply:
@azayati Could you check now ?